changelog/0032-upgrade-elasticsearch.yml

```diff
@@ -0,0 +1,76 @@
+---
+title: ES upgrade from 1.7.6 to 2.4.6
+key: upgrade-elasticsearch
+date: 2020-02-05
+optional_per_env: yes
+update_steps: |
+
+  1. Stop the site
+  ```bash
+  commcare-cloud <env> downtime start
+  ```
+    
+  2. Create a PR by adding below parameters in commcare-cloud/environments/<env>/public.yml and merge
+  { 
+  elasticsearch_version: 2.4.6
+  elasticsearch_download_sha256: 5f7e4bb792917bb7ffc2a5f612dfec87416d54563f795d6a70637befef4cfc6f.
+  ELASTICSEARCH_MAJOR_VERSION: 2
+  }
+    
+  3. Update the local settings
+  ```bash
+  commcare-cloud <env> update-config
+  ```  
+  4. Login to any ES server and disable the cluster routing
+  ```bash
+  curl -XPUT Elasticsearchserver-IP:9200/_cluster/settings -d '{\"persistent\" : {\"cluster.routing.allocation.enable\" : \"none\" }}'
+  ```
+    
+  5. Stop the monit and Elasticsearch service
+  ```bash
+  commcare-cloud <env> run-shell-command all "service monit stop" -b --limit=elasticsearch
+  commcare-cloud <env> run-shell-command all "service elasticsearch stop" -b --limit=elasticsearch
+  ```
+    
+  6. Download and extract the new version with below command and which will start the ES service with new version also
+  ```bash
+  commcare-cloud <env> ansible-playbook deploy_stack.yml --limit=elasticsearch --tags=elasticsearch
+  ```
+    
+  7. Stop the monit and Elasticsearch service
+  ```bash
+  commcare-cloud <env> run-shell-command all "service monit stop" -b --limit=elasticsearch
+  commcare-cloud <env> run-shell-command all "service elasticsearch stop" -b --limit=elasticsearch
+  ```
+    
+  8. Rename the 1.7.6 ES data folder with new version 2.4.6
+  ```bash
+  commcare-cloud <env> run-shell-command all "mv /opt/data/elasticsearch-2.4.6 /opt/data/elasticsearch-2.4.6-new-installation" -b --limit=elasticsearch
+  commcare-cloud <env> run-shell-command all "mv /opt/data/elasticsearch-1.7.6 /opt/data/elasticsearch-2.4.6" -b --limit=elasticsearch
+  ```
```

Comment:
@sanjay2916 Shouldn't we be telling people to first copy the data and _then_ install the new version of Elasticsearch (move this step to before step 6)? Then we wouldn't need step 7, and we wouldn't need to `mv /opt/data/elasticsearch-2.4.6 /opt/data/elasticsearch-2.4.6-new-installation` because `/opt/data/elasticsearch-2.4.6` wouldn't exist yet.

Reply:
@dannyroberts  After we run deploy stack , the es 2.4.6 empty directory will create in /opt/data/  folder and if you move 1.7.6 to 2.4.6 before renaming of 2.4.6 directory it will go inside the 2.4.6 directory

Reply:
Hey @sanjay2916, I was asking something different. I was asking, why not use the following steps?

1. stop elasticsearch
2. `cp -r /opt/data/elasticsearch-1.7.6 /opt/data/elasticsearch-2.4.6`
3. Verify permissions and size of data directory
4. Install new elasticsearch with `commcare-cloud <env> ansible-playbook deploy_stack.yml --limit=elasticsearch --tags=elasticsearch` (which also starts elasticsearch automatically)
5. Verify the cluster status is yellow
6. ...

Was there some reason to move the data _after_ updating ES instead of before it?

Regardless, since you've tested it many times the way you have it here, maybe it's best to keep it this way rather than asking people to do something that's not well-tested.